src/pages/EnterRoom.vue

```diff
@@ -0,0 +1,84 @@
+<script setup lang="ts">
+import { ref, onMounted } from 'vue'
+import api, { EnterRoom } from '@/lib/apis'
+import router from '@/router'
+
+import { useStoreUser } from '@/stores/user'
+const store = useStoreUser()
+// 部屋に入る前のページ
+// 必須: 名前
+// private -> 合言葉の入力欄
+// 名前を適切に設定 && 合言葉が正しい -> IndividualRoom へ
+// 名前に被り || 合言葉が不正 -> 失敗を表示して再入力へ
+
+const thisRoomId = ref('')
+const userNickName = ref('')
+const roomPassword = ref('')
+const submitError = ref(false)
+const userSettingError = ref(false)
+const submit = async () => {
+  const enterInfo: EnterRoom = {
+    userName: userNickName.value,
+    password: roomPassword.value,
+  }
+  try {
+    // enterRoom を送る
+    const resp = await api.apiRoomRoomIdEnterPost(thisRoomId.value, enterInfo)
+    // ここで resp から userId と userName を持たせて IndividualRoom へ
+    if (resp.status == 200) {
+      submitError.value = false
+      // stores/user に userId と userName を記録して、 /rooms/:id に移動させる
+      userNickName.value = resp.data.userName
+      const userId = resp.data.userId
+      store.setUser(userNickName.value, userId)
```

Comment:
`userNickName` はユーザーからの入力受付用なので、サーバーからのレスポンスを入れるのはあまり良くないかも

```suggestion
      const userId = resp.data.userId
      store.setUser(resp.data.userName, userId)
```